COQ_USE_DUNE affects the build of coq through opam

#### Description of the problem

If `COQ_USE_DUNE` is set, then the `opam` installation of `coq.dev` will try to use dune and end up not installing anything ~~while believing it was successful.~~

Steps to reproduce:

```shell
export COQ_USE_DUNE=true
opam switch create test 4.12.0
opam update
opam repo add coq-core-dev https://coq.inria.fr/opam/core-dev
opam install coq.dev -vv # observe dune is being used and an error is produced
```
Cf. the [zulip discussion](https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs.20.26.20plugin.20devs/topic/Installing.20coq.2Edev.20through.20opam.2C.20missing.20binaries).

If installing `coq.8.13.2` instead, `opam` still tries to use `dune`, but it fails with `make: *** No rule to make target 'byte'.`. It's the same underlying problem, ~~but at least you are alerted that something went wrong.~~ EDIT: after #14466 there is also an error with `coq.dev`.

#### Coq Version

Any version, although `COQ_USE_DUNE` is a recent addition to `master` as far as I know.

--- a/Makefile
+++ b/Makefile
@@ -12,5 +12,9 @@
 ifndef COQ_USE_DUNE
 include Makefile.make
 else
+ifdef OPAM_PACKAGE_NAME # installing through opam: ignore COQ_USE_DUNE
+include Makefile.make
+else
 include Makefile.dune
+endif
 endif

